### DISCUSSION TITLE
Access Pre Run Results

### DISCUSSION THREAD

**Discussion by nicdichiara** [thunderclient/thunder-client-support#862] [General]
Hi there. Thank you for the awesome tool, Thunderclient. 

I love the new feature Pre Run - But cannot figure out how to access data retrieved from the Pre Run? I cannot seem to find find instructions in the docs or anywhere. 

For example, I would like to use the Pre Run to get a token, and then utilize that token in the following request. Is it possible to address that returned data? 

Thanks!
-Nic

**Comment by rangav** [comment]
Yea it’s possible, in the pre-request use the tests tab and select Set Env variable to save token from response 

and use that token variable in the main request

**Reply by nicdichiara** [reply]
Great - thank you! Works perfectly!
Awesome feature.

**Reply by rangav** [reply]
Thanks, Glad you like the feature :)

**Reply by rangav** [reply]
`Set Environment variable` documentation is here
https://github.com/rangav/thunder-client-support#setenv

**Reply by nicdichiara** [reply]
Thank you - I was searching for "Pre Run" and skipped over that section.

**Reply by rangav** [reply]
I haven't added any documentation for `pre-run`, feel free to update docs if you have any suggestions.
